lib/appium/AppiumRunner.js

```diff
@@ -0,0 +1,483 @@
+        if (files && files.length > 0) {
+            logger.normal("paramedic-appium: Found app package: " + files[0]);
+            return files[0];
+        }
+        throw new Error("Could not find the app package");
+    }
+}
+
+function getPluginDirs(appPath) {
+    return shell.ls(path.join(appPath, "/plugins/cordova-plugin-*"));
+}
+
+function getConfigPath(appPath) {
+    return path.join(appPath, "config.xml");
+}
+
+function parseElementtreeSync(filename) {
+    var contents = fs.readFileSync(filename, util.DEFAULT_ENCODING);
+    if (!contents) {
+        throw new Error("The config file is empty: " + filename);
+    }
+    // Skip the Byte Order Mark (BOM)
+    contents = contents.substring(contents.indexOf("<"));
+
+    return new elementTree.ElementTree(elementTree.XML(contents));
+}
+
+function addCspSource(appPath, directive, source) {
+    var cspInclFile = path.join(appPath, "www/csp-incl.js");
+    var indexFile = path.join(appPath, "www/index.html");
+    var cspFile = fs.existsSync(cspInclFile) ? cspInclFile : indexFile;
+    var cspContent = fs.readFileSync(cspFile, util.DEFAULT_ENCODING);
+    var cspTagOpening = "<meta http-equiv=\"Content-Security-Policy\" content=\"";
+    var cspRule = directive + " " + source;
+    var cspRuleReg = new RegExp(directive + "[^;\"]+" + source.replace("*", "\\*"));
+
+    logger.normal("paramedic-appium: Adding CSP source \"" + source + "\" to directive \"" + directive + "\"");
+
+    if (cspContent.match(cspRuleReg)) {
+        logger.normal("paramedic-appium: It's already there.");
+    } else if (util.contains(cspContent, directive)) {
+        // if the directive is there, just add the source to it
+        cspContent = cspContent.replace(directive, cspRule);
+        fs.writeFileSync(cspFile, cspContent, util.DEFAULT_ENCODING);
+    } else if (cspContent.match(/content=".*?default-src.+?"/)) {
+        // needed directive is not there but there is default-src directive
+        // creating needed directive and copying default-src sources to it
+        var defaultSrcReg = /(content=".*?default-src)(.+?);/;
+        cspContent = cspContent.replace(defaultSrcReg, "$1$2; " + cspRule + "$2;");
+        fs.writeFileSync(cspFile, cspContent, util.DEFAULT_ENCODING);
+    } else if (util.contains(cspContent, cspTagOpening)) {
+        // needed directive is not there and there is no default-src directive
+        // but the CSP tag is till present
+        // just adding needed directive to a start of CSP tag content
+        cspContent = cspContent.replace(cspTagOpening, cspTagOpening + directive + " " + source + "; ");
+        fs.writeFileSync(cspFile, cspContent, util.DEFAULT_ENCODING);
+    } else {
+        // no CSP tag, skipping
+        logger.normal("paramedic-appium: WARNING: No CSP tag found.");
+    }
+}
+
+function setPreference(appPath, preference, value) {
+    var configFile = getConfigPath(appPath);
+    var xml = parseElementtreeSync(configFile);
```

Comment:
Can we just reuse `cordova-common`'s `ConfigParser` to access config's preferences?